controllers/front/validation.php

```diff
@@ -98,122 +98,135 @@ public function postProcess()
+        $sql = 'SELECT * FROM '._DB_PREFIX_."blockonomics_bitcoin_orders WHERE id_cart = $cart->id";
+        $order = Db::getInstance()->getRow($sql);
 
-        $current_time = time();
-        $price = $blockonomics->getBTCPrice($currency->id);
-
-        if (!$price) {
-            Tools::redirectLink(__PS_BASE_URI__ . 'order.php?step=1');
-        }
+        if(!$order || $order['value'] != $total) {
+            $current_time = time();
+            $bits = $this->get_bits($blockonomics, $currency, $total);
+            $time_remaining = Configuration::get('BLOCKONOMICS_TIMEPERIOD');
+            $responseObj = $blockonomics->getNewAddress();
+            if (!$responseObj->data || !$responseObj->data->address) {
+                $this->displayError($blockonomics);
+            }
+            $address = $responseObj->data->address;
 
-        $bits = (int) ((1.0e8 * $total) / $price);
+            // Create backup cart
+            $old_cart_secure_key = $cart->secure_key;
+            $old_cart_customer_id = (int)$cart->id_customer;
+            $cart_products = $cart->getProducts();
+            $new_cart = new Cart();
+            $new_cart->id_lang = $this->context->language->id;
+            $new_cart->id_currency = $this->context->currency->id;
+            $new_cart->add();
+            foreach ($cart_products as $product) {
+                $new_cart->
+                    updateQty(
+                        (int) $product['quantity'],
+                        (int) $product['id_product'],
+                        (int) $product['id_product_attribute']
+                    );
+            }
+            if ($this->context->cookie->id_guest) {
+                $guest = new Guest($this->context->cookie->id_guest);
+                $new_cart->mobile_theme = $guest->mobile_theme;
+            }
 
-        if (!extension_loaded('intl')) {
-            $this->displayExtError($blockonomics);
-        }
-
-        // Create backup cart
-        $old_cart_secure_key = $cart->secure_key;
-        $old_cart_customer_id = (int)$cart->id_customer;
-        $cart_products = $cart->getProducts();
-        $new_cart = new Cart();
-        $new_cart->id_lang = $this->context->language->id;
-        $new_cart->id_currency = $this->context->currency->id;
-        $new_cart->add();
-        foreach ($cart_products as $product) {
-            $new_cart->
-                updateQty(
-                    (int) $product['quantity'],
-                    (int) $product['id_product'],
-                    (int) $product['id_product_attribute']
-                );
-        }
-        if ($this->context->cookie->id_guest) {
-            $guest = new Guest($this->context->cookie->id_guest);
-            $new_cart->mobile_theme = $guest->mobile_theme;
-        }
+            // Validate the order
+            $mailVars = array(
+                '{bitcoin_address}' => $address,
+                '{bits}' => $bits / 1.0e8,
+                '{track_url}' =>
```

Comment:
track_url should be removed during https://github.com/blockonomics/prestashop-plugin/pull/90. Please confirm this is not required and remove it accordingly.

Reply:
I've tested creating a new order after removing `track_url`. The checkout works fine without it. However, it looks like `track_url` is a part of the email that should go out to the customer after placing an order? Is there a way of testing or looking on an outgoing email with and without `track_url`?

This branch is based on the 1.7 branch after it was merged with #90. There should be no merge conflicts after removing `track_url`. Just to want to make sure that the outgoing email still looks like it should without `track_url`.

Reply:
Test outgoing mails from Prestashop by setting up your mail server/smtp info from the Prestashop Admin > Advanced Parameters > Email.
Order confirmation emails use the order_conf.html template which does not include track_url.

Reply:
Thanks, now I've tested the email after removing `track_url` and it works, just as you said. Just feels better to give it a go to make sure.